src/acp/policy.test.ts

```diff
@@ -0,0 +1,254 @@
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { describe, it, expect } from "@jest/globals";
+jest.mock("../fetcher.ts", () => ({
+  fetch: jest.fn().mockImplementation(() =>
+    Promise.resolve(
+      new Response(undefined, {
+        headers: { Location: "https://arbitrary.pod/resource" },
+      })
+    )
+  ),
+}));
+
+import { Response } from "cross-fetch";
+import { rdf, acp } from "../constants";
+import { createSolidDataset } from "../resource/solidDataset";
+import { getUrl, getUrlAll } from "../thing/get";
+import { removeUrl } from "../thing/remove";
+import { setUrl } from "../thing/set";
+import {
+  asUrl,
+  createThing,
+  getThing,
+  getThingAll,
+  setThing,
+} from "../thing/thing";
+import {
+  createPolicy,
+  getPolicy,
+  getPolicyAll,
+  removePolicy,
+  savePolicyDatasetAt,
+  setPolicy,
+} from "./policy";
+
+const policyUrl = "https://some.pod/policy-resource";
+
+describe("savePolicyDatasetAt", () => {
+  it("sets the type of acp:AccessPolicy if not set yet", async () => {
+    const mockFetch = jest.fn(window.fetch).mockResolvedValue(new Response());
+    const newDataset = createSolidDataset();
+
+    const savedDataset = await savePolicyDatasetAt(policyUrl, newDataset, {
+      fetch: mockFetch,
+    });
+
+    const savedDatasetThing = getThing(savedDataset, policyUrl);
+    expect(savedDatasetThing).not.toBeNull();
+    expect(getUrl(savedDatasetThing!, rdf.type)).toBe(acp.AccessPolicyResource);
+  });
+
+  it("overwrites an existing type that might be set", async () => {
+    const mockFetch = jest.fn(window.fetch).mockResolvedValue(new Response());
+    let newDatasetThing = createThing({ url: policyUrl });
+    newDatasetThing = setUrl(
+      newDatasetThing,
+      rdf.type,
+      "https://arbitrary.vocab/ArbitraryClass"
+    );
+    const newDataset = setThing(createSolidDataset(), newDatasetThing);
+
+    const savedDataset = await savePolicyDatasetAt(policyUrl, newDataset, {
+      fetch: mockFetch,
+    });
+
+    const savedDatasetThing = getThing(savedDataset, policyUrl);
+    expect(savedDatasetThing).not.toBeNull();
+    expect(getUrlAll(savedDatasetThing!, rdf.type)).toEqual([
+      acp.AccessPolicyResource,
+    ]);
+  });
+
+  it("calls the included fetcher by default", async () => {
+    const mockedFetcher = jest.requireMock("../fetcher.ts") as {
+      fetch: jest.Mock<
+        ReturnType<typeof window.fetch>,
+        [RequestInfo, RequestInit?]
+      >;
+    };
+
+    await savePolicyDatasetAt(policyUrl, createSolidDataset());
+
+    expect(mockedFetcher.fetch.mock.calls[0][0]).toBe(policyUrl);
+  });
+});
```

Comment:
There is no test checking explicitly that the provided fetch is used.

Reply:
Yeah fair enough: 097f37f351974e50b00bf884e3d12e372f5add3d